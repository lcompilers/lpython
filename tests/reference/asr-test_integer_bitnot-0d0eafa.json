{
    "basename": "asr-test_integer_bitnot-0d0eafa",
    "cmd": "lpython --show-asr --indent --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/test_integer_bitnot.py",
    "infile_hash": "ce63509c56eb6c68ca6d64cc2a195644f3d2fc285f3ec8865551eff5",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-test_integer_bitnot-0d0eafa.stdout",
    "stdout_hash": "dcde5e01a94adf2a43802f42a3571ed47471faee31b25cf5c8dc443c",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}