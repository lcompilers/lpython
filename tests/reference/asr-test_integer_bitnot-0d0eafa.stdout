(TranslationUnit
    (SymbolTable
        1
        {
            _global_symbols:
                (Module
                    (SymbolTable
                        5
                        {
                            _lpython_main_program:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            
                                        })
                                    _lpython_main_program
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [f]
                                    []
                                    [(SubroutineCall
                                        5 f
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                ),
                            f:
                                (Function
                                    (SymbolTable
                                        2
                                        {
                                            i:
                                                (Variable
                                                    2
                                                    i
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            res:
                                                (Variable
                                                    2
                                                    res
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    f
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(=
                                        (Var 2 i)
                                        (IntegerConstant 5 (Integer 4 []))
                                        ()
                                    )
                                    (=
                                        (Var 2 res)
                                        (IntegerBitNot
                                            (Var 2 i)
                                            (Integer 4 [])
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (Var 2 res)
                                            Eq
                                            (IntegerUnaryMinus
                                                (IntegerConstant 6 (Integer 4 []))
                                                (Integer 4 [])
                                                (IntegerConstant -6 (Integer 4 []))
                                            )
                                            (Logical 4 [])
                                            ()
                                        )
                                        ()
                                    )
                                    (=
                                        (Var 2 i)
                                        (IntegerUnaryMinus
                                            (IntegerConstant 235346 (Integer 4 []))
                                            (Integer 4 [])
                                            (IntegerConstant -235346 (Integer 4 []))
                                        )
                                        ()
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (IntegerBitNot
                                                (Var 2 i)
                                                (Integer 4 [])
                                                ()
                                            )
                                            Eq
                                            (IntegerConstant 235345 (Integer 4 []))
                                            (Logical 4 [])
                                            ()
                                        )
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                )
                        })
                    _global_symbols
                    []
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        3
                        {
                            _lpython_main_program:
                                (ExternalSymbol
                                    3
                                    _lpython_main_program
                                    5 _lpython_main_program
                                    _global_symbols
                                    []
                                    _lpython_main_program
                                    Public
                                )
                        })
                    main_program
                    [_global_symbols]
                    [(SubroutineCall
                        3 _lpython_main_program
                        ()
                        []
                        ()
                    )]
                )
        })
    []
)
